app/store/eventSlice.ts

```diff
@@ -3,8 +3,17 @@ import { EventState } from '~/utils/interfaces';
 
 export const createEventSlice: StateCreator<EventState, [], [], EventState> = (set) => ({
   events: [],
-  setEvents: (events) => set((state) => ({ events: [...events] })),
+  setEvents: (events) => set(() => ({ events: [...events] })),
   addEvent: (event) => set((state) => ({ events: [...state.events, event] })),
   removeEvent: (event) =>
     set((state) => ({ events: state.events.filter(({ id }) => id !== event.id) })),
+  updateEvent: (event) =>
+    set((state) => ({
+      events: state.events.map((e) => {
+        if (e.id === event.id) {
+          return event;
+        }
+        return e;
+      }),
```

Comment:
can be written like this for better readability and understanding:
```js
events: state.events.map((e) => (e.id === event.id ? event : e)),
```

Reply:
updating it in the next commits